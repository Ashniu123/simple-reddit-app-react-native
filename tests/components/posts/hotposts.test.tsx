import * as React from 'react';
import { shallow } from 'enzyme';

import { HotPost } from '../../../src/components/posts/hotposts.component';

let startFetchPosts, navigation;

beforeEach(() => {
	startFetchPosts = jest.fn();
	navigation = { navigate: jest.fn() }
});

test('should render HotPost component correctly', () => {
	const wrapper = shallow(<HotPost 
		startFetchPosts={startFetchPosts}
		navigation={navigation}
	/>);
	expect(wrapper).toMatchSnapshot();
});

test('should startFetchPosts with "hot" if token exists and timestamp difference is lesser than 1 hr', () => {
	const token = 'Some Token';
	const timestamp = Date.now() + 3600;
	const wrapper = shallow(<HotPost
		startFetchPosts={startFetchPosts} 
		navigation={navigation}
		timestamp={timestamp}
		token={token}
	/>);
	expect(startFetchPosts).toHaveBeenCalledWith('hot');
	expect(navigation.navigate).not.toHaveBeenCalled();
});

test('should not startFetchPosts with "hot" if token exists and timestamp difference is greater than 1 hr', () => {
	const token = 'Some Token';
	const timestamp = 0; //beginning of UTC time :D
	const wrapper = shallow(<HotPost
		startFetchPosts={startFetchPosts}
		navigation={navigation}
		timestamp={timestamp}
		token={token}
	/>);
	expect(startFetchPosts).not.toHaveBeenCalled();
	expect(navigation.navigate).toHaveBeenCalledWith('Auth');
});

test('should not startFetchPosts with "hot" if token does not exists and timestamp difference is lesser than 1 hr', () => {
	const token = '';
	const timestamp = Date.now() + 3600;
	const wrapper = shallow(<HotPost
		startFetchPosts={startFetchPosts}
		navigation={navigation}
		timestamp={timestamp}
		token={token}
	/>);
	expect(startFetchPosts).not.toHaveBeenCalled();
	expect(navigation.navigate).toHaveBeenCalledWith('Auth');
});