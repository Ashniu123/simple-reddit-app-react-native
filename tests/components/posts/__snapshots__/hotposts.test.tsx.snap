// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HotPost component correctly 1`] = `
<FlatList
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
    />
  }
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
